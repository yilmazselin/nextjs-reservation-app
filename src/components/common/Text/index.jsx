import styled from "styled-components"
import { font } from "../../../utils/helper"

const Text = styled.p`
  ${(props) => `
    ${font({
    size: props.fontSize,
    weight: props.fontWeight,
    color: props.color
  })}
  `};
  ${(props) => ({ ...props.params })};
`

export default Text
