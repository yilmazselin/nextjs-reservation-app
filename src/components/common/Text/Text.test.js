import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import "jest-styled-components"
import Text from "."

describe("<Text />", () => {
  it("should render correctly", () => {
    const { container } = render(<Text children="Hello" />)

    expect(container).toMatchSnapshot()
  })

  it("should have custom CSSProps", () => {
    render(<Text
      children="Hello"
      params={{ color: "red",
        fontSize: 24 }}
    />)
    const text = screen.getByText("Hello")

    expect(text).toHaveStyle({
      color: "red",
      fontSize: 24
    })
  })
})
