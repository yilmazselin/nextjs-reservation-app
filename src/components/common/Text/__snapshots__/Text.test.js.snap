// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render correctly 1`] = `
.c0 {
  color: #E82223;
  font-size: 14px;
  font-family: "NunitoSans-400","sans-serif";
}

<div>
  <p
    class="c0"
  >
    Hello
  </p>
</div>
`;
